octoprint_octorelay/__init__.py

```diff
@@ -196,8 +196,11 @@ def toggle_relay(self, index, target: Optional[bool] = None):
             f"Toggling relay {index} on pin {pin}" if target is None else
             f"Turning the relay {index} {'ON' if target else 'OFF'} (pin {pin})"
         )
-        cmd = settings["cmd_on" if relay.toggle(target) else "cmd_off"]
+        state = relay.toggle(target)
+        cmd = settings["cmd_on" if state else "cmd_off"]
         self.run_system_command(cmd)
+        if state:
+            self.handle_plugin_event(TURNED_ON, scope = [index])
```

Comment:
- [ ] here is another potential issue: infinite loops: should prevent `relay.state=target` (same state) toggling or it's side effects

Reply:
OR! I can forbid to event TURNED_ON having action ON